pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -56,9 +56,396 @@ stdenv.mkDerivation rec {
+      linker_flag: "-undefined"
+      linker_flag: "dynamic_lookup"
+      linker_flag: "-headerpad_max_install_names"
+      # We know all files in `/nix/store` are immutable so it's safe to disable inclusion checks for them
+      cxx_builtin_include_directory: "/nix/store"
+      tool_path { name: "gcov" path: "${coreutils}/bin/false" }
+      tool_path { name: "ld" path: "${cctools}/bin/ld" }
+      tool_path { name: "nm" path: "${cctools}/bin/nm" }
+      tool_path { name: "objcopy" path: "${binutils}/bin/objcopy" }
+      objcopy_embed_flag: "-I"
+      objcopy_embed_flag: "binary"
+      tool_path { name: "objdump" path: "${binutils}/bin/objdump" }
+      tool_path { name: "strip" path: "${cctools}/bin/strip" }
+
+      # Anticipated future default.
+      unfiltered_cxx_flag: "-no-canonical-prefixes"
+
+      # Make C++ compilation deterministic. Use linkstamping instead of these
+      # compiler symbols.
+      unfiltered_cxx_flag: "-Wno-builtin-macro-redefined"
+      unfiltered_cxx_flag: "-D__DATE__=\"redacted\""
+      unfiltered_cxx_flag: "-D__TIMESTAMP__=\"redacted\""
+      unfiltered_cxx_flag: "-D__TIME__=\"redacted\""
+
+      # Security hardening on by default.
+      # Conservative choice; -D_FORTIFY_SOURCE=2 may be unsafe in some cases.
+      compiler_flag: "-D_FORTIFY_SOURCE=1"
+      compiler_flag: "-fstack-protector"
+
+      # Enable coloring even if there's no attached terminal. Bazel removes the
+      # escape sequences if --nocolor is specified.
+      compiler_flag: "-fcolor-diagnostics"
+
+      # All warnings are enabled. Maybe enable -Werror as well?
+      compiler_flag: "-Wall"
+      # Enable a few more warnings that aren't part of -Wall.
+      compiler_flag: "-Wthread-safety"
+      compiler_flag: "-Wself-assign"
+
+      # Keep stack frames for debugging, even in opt mode.
+      compiler_flag: "-fno-omit-frame-pointer"
+
+      # Anticipated future default.
+      linker_flag: "-no-canonical-prefixes"
+
+      compilation_mode_flags {
+        mode: DBG
+        # Enable debug symbols.
+        compiler_flag: "-g"
+      }
+      compilation_mode_flags {
+        mode: OPT
+        # No debug symbols.
+        # Maybe we should enable https://gcc.gnu.org/wiki/DebugFission for opt or even generally?
+        # However, that can't happen here, as it requires special handling in Bazel.
+        compiler_flag: "-g0"
+
+        # Conservative choice for -O
+        # -O3 can increase binary size and even slow down the resulting binaries.
+        # Profile first and / or use FDO if you need better performance than this.
+        compiler_flag: "-O2"
+
+        # Disable assertions
+        compiler_flag: "-DNDEBUG"
+
+        # Removal of unused code and data at link time (can this increase binary size in some cases?).
+        compiler_flag: "-ffunction-sections"
+        compiler_flag: "-fdata-sections"
+      }
+      linking_mode_flags { mode: DYNAMIC }
+    }
+  '' + lib.optionalString stdenv.isLinux ''
```

Comment:
Essentially the same file as above as far as I can see (it’s hard to diff by scrolling, but it looks like it is mostly a duplicate).